security/rbac/enforcer.go

```diff
@@ -0,0 +1,98 @@
+package rbac
+
+import (
+	"strings"
+
+	"github.com/gaia-pipeline/gaia"
+)
+
+// PolicyEnforcer is for enforcing RBAC policies.
+type PolicyEnforcer interface {
+	Enforce(policyNames []string, namespace gaia.RBACPolicyNamespace, action gaia.RBACPolicyAction) bool
+	ResolvePolicies(policyNames []string) namespaceActionMap
+}
+
+type policyEnforcer struct {
+	svc Service
+}
+
+type namespaceActionMap map[gaia.RBACPolicyNamespace]map[gaia.RBACPolicyAction]interface{}
+
+// NewPolicyEnforcer creates a new policyEnforcer
+func NewPolicyEnforcer(svc Service) PolicyEnforcer {
+	return &policyEnforcer{svc: svc}
+}
+
+// Enforce takes a list of policy names from a Gaia users JWT claims, a required namespace and action. We get all the
+// users policies using the names provided. We then merged all the policies together into the namespaceActionMap which
+// acts as a quick lookup for the namespace and action that is being enforced.
+func (s *policyEnforcer) Enforce(policyNames []string, namespace gaia.RBACPolicyNamespace, action gaia.RBACPolicyAction) bool {
+	resolved := s.ResolvePolicies(policyNames)
+
+	if ns, nsOk := resolved[namespace]; nsOk {
+		// first, check for a wildcard.
+		if _, acOk := ns["*"]; acOk {
+			return true
+		}
+		// second, check for the specific action.
+		if _, acOk := ns[action]; acOk {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (s *policyEnforcer) ResolvePolicies(policyNames []string) namespaceActionMap {
+	na := make(namespaceActionMap)
+
+	// iterate through the user policy names provided
+	for _, policy := range policyNames {
+		// get the policy from the service/cache
+		policyResource, _ := s.svc.GetPolicy(policy)
```

Comment:
Do check for errors here though... If the policy doesn't exists, this should throw some kind of warning.